src/gateway/router/routes/interactionsRoute.ts

```diff
@@ -1,108 +1,7 @@
-  to && bindings.push(to as string);
-  source && bindings.push(source as string);
-  upToTransactionId && bindings.push(upToTransactionId as string);
-  parsedPage && bindings.push(parsedLimit);
-  parsedPage && bindings.push(offset);
-
-  try {
-    const result: any = await gatewayDb.raw(
-      `
-          SELECT interaction, 
-                 confirmation_status, 
-                 ${shouldMinimize ? '' : 'confirming_peer, confirmations, bundler_tx_id, '} 
-                 count(*) OVER () AS total
-          FROM interactions 
-            WHERE (contract_id = ? OR interact_write @> ARRAY[?]) 
-          ${
-            parsedConfirmationStatus
-              ? ` AND confirmation_status IN (${parsedConfirmationStatus.map((status) => `'${status}'`).join(', ')})`
-              : ''
-          } 
-          ${from ? ' AND block_height >= ?' : ''} 
-          ${to ? ' AND block_height <= ?' : ''} 
-          ${source ? `AND source = ?` : ''} 
-          ${upToTransactionId ? `AND id <= (SELECT id FROM interactions WHERE interaction_id = ?)` : ''} 
-          ORDER BY block_height DESC, interaction_id DESC ${parsedPage ? ' LIMIT ? OFFSET ?' : ''};
-      `,
-      bindings
-    );
-
-    const totalInteractions: any =
-      totalCount == 'true' &&
-      (await gatewayDb.raw(
-        `
-          SELECT count(case when confirmation_status = 'corrupted' then 1 else null end)     AS corrupted,
-                 count(case when confirmation_status = 'confirmed' then 1 else null end)     AS confirmed,
-                 count(case when confirmation_status = 'not_processed' then 1 else null end) AS not_processed,
-                 count(case when confirmation_status = 'forked' then 1 else null end)        AS forked
-          FROM interactions
-          WHERE contract_id = ?;
-      `,
-        contractId
-      ));
-
-    const total = result?.rows?.length > 0 ? result?.rows[0].total : 0;
-
-    ctx.body = {
-      paging: {
-        total,
-        limit: parsedLimit,
-        items: result?.rows.length,
-        page: parsedPage,
-        pages: Math.ceil(total / parsedLimit),
-      },
-      ...(totalInteractions && {
-        total: {
-          confirmed: totalInteractions?.rows[0].confirmed,
-          corrupted: totalInteractions?.rows[0].corrupted,
-          not_processed: totalInteractions?.rows[0].not_processed,
-          forked: totalInteractions?.rows[0].forked,
-        },
-      }),
-      interactions: result?.rows?.map((r: any) => ({
-        status: r.confirmation_status,
-        confirming_peers: r.confirming_peer,
-        confirmations: r.confirmations,
-        interaction: {
-          ...r.interaction,
-          bundlerTxId: r.bundler_tx_id,
-        },
-      })),
-    };
-  } catch (e: any) {
-    ctx.logger.error(e);
-    ctx.status = 500;
-    ctx.body = { message: e };
-  }
+  ctx.logger.error('Route deprecated. /v2/interactions-sort-key should be used instead.');
+  ctx.status = 410;
```

Comment:
not sure what to do with this one, from my perspective it is impossible to provide a proper error body message to the old versions of SDK.

Reply:
2022-12-06T09:59:19.404Z ERROR [deploy] Error: Unable to retrieve transactions. Warp gateway responded with status 410. []

Alternatively, I can just leave the old block-height endpoint with optimizations.

Reply:
Yeah, I guess we need to:
1. leave them for now with optimizations
2. warn on twitter and discord that in about a month they will be closed and people need to upgrade to warp
3. after ~ month - return error from this endpoint.